OneCommunityGlobal/HighestGoodNetworkApp
# Jianjun - all three total report feature

# Description (7/22 Update)
**(PRIORITY HIGH) Jae: Create “Tasks Summary” on the Reports page**
**Note here: to avoid confusion, change 'task' to 'project' in this feature.**
Admin/Owner Login → Reports → Reports
In the view below, I need a “Total Tasks Report” button to see exactly how many new tasks have been worked on for a designated time period
Tasks must have had at least 1 hour logged to them to count 
Should be able to choose the period of time and the result should show on this page or as a Modal
Should include a “Total Hours” section showing total tangible time logged to all tasks during the selected period
Would like to run a “details” report that will list all the tasks too, with clickable names to visit their details
Need an “i” by the button to explain what the button does (all of the above)
This is needed because it is a metric we’re using from here on forward to measure the growth and progress of our nonprofit. So I want to be able to easily run this for individual months and/or an entire fiscal year to compare results with previous years.
A bar graph visualization of each month when seeing the yearly summary would be super helpful too

**(PRIORITY HIGH) Jae: Create “People Summary” on the Reports page**
Admin/Owner Login → Reports → Reports
In the view below, I need a “Total Team’s Report” button to see exactly how many total teams have contributed time to the project for a designated time period
Teams must have had at least 10 hours logged for them to count/be included
Should be able to choose the period of time and the result should show on this page or as a Modal
Should include a “Total Hours” section showing total tangible time logged to all teams during the selected period
Would like to run a “details” report that will list all the teams and hours contributed for each team for that time period too, with clickable names that will show who was on each team
Need an “i” by the button to explain what the button does (all of the above)
This is needed because it is a metric we’re using from here on forward to measure the growth and progress of our nonprofit. So I want to be able to easily run this for individual months and/or an entire fiscal year to compare results with previous years.
A bar graph visualization of each month when seeing the yearly summary would be super helpful too

**(PRIORITY HIGH) Jae: Create “Teams Summary” on the Reports page**
Admin/Owner Login → Reports → Reports
In the view below, I need a “Total People Report” button to see exactly how many total people have contributed time to the project for a designated time period
People must have had at least 10 hours logged for them to count/be included
Should be able to choose the period of time and the result should show on this page or as a Modal
Should include a “Total Hours” section showing total tangible time logged by all team members during the selected period
Would like to run a “details” report that will list all the people and hours contributed by each for that time period too, with clickable names to visit their profiles
Need an “i” by the button to explain what the button does (all of the above)
This is needed because it is a metric we’re using from here on forward to measure the growth and progress of our nonprofit. So I want to be able to easily run this for individual months and/or an entire fiscal year to compare results with previous years.
A bar graph visualization of each month when seeing the yearly summary would be super helpful too

## Main changes explained:
- Add three buttons that can hide or show the total report with data tips.
- For each feature, create the information card that shows the number of total qualified teams/people (with more than 10 hours) or projects (with more than 1 hour) and the total tangible time logged in this period.
- Two bar graph to show the number of the valid projects/members/teams monthly or yearly.
- A detail table listing all the teams/members/projects and the hours in the period.
- The detail table can show the list of team members or direct to the user profile page or project report page by clicking the team/user/project name.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. Admin/Owner Login → Reports → Reports
5. try to select different start/end date and then click the three buttons.

## Screenshots or videos of changes:
<img width="497" alt="1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/1752152c-805e-4764-b079-ae181782d8d5">
<img width="494" alt="2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/0383cb42-052c-44f4-80d7-ae6f430d8f8a">
<img width="490" alt="3" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/4e0ab343-1d50-4708-aded-e29406ca6a38">
<img width="502" alt="4" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/5edf0f06-f00f-4a1f-9331-5c8d8fa5688b">
<img width="494" alt="5" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/a6a75265-af33-4fa5-baaf-1ca91ff639e4">

## Note:
Note that the two bar graph will not always show in this section. It depends on the length of the time period you select.
Now the conditions are as following:
The time period is less than one month: no bar graph.
between one month to about twelve month: show monthly bar graph.
The start date and the end date are in different year: show yearly bar graph.
